### PR TITLE
Rewrite and simplify the assigner code.

### PR DESCRIPTION
**Pull request by spetrovic77**
In particular, we now store all of the assigner state in a series of ReplicaSetState entries in the store, and find application and application-version subsets of those using the `store.List` operation.

Additionally, all Pods in a ReplicaSet are discovered, health- checked, and load-inspected entirely by the assigner, periodically. This simplifies the state update at the Assigner significantly.

Other changes:
  * Unify health-checking for routed and non-routed components back at the assigner.
  * A small number of cleanups throughout the code.

### CODE REVIEW

File: internal/local/starter.go [hunk 1]
```diff
@@ -100,44 +102,46 @@ func (s *Starter) Start(ctx context.Context, cfg *config.GKEConfig, replicaSetNa
...
+	}
+	rs, ok := v.replicaSets[replicaSet]
+	if !ok { // replica set hasn't yet started
+		return nil, nil
+	}
+	var healthyPods []*nanny.Pod
+	for _, r := range rs.pods {
+		babysitterAddr := r.b.SelfAddr()
+		weaveletAddr := r.b.WeaveletInfo().DialAddr
+		b := newBabysitter(babysitterAddr)
```
Comment (mwhittaker) [inline internal/local/starter.go:132]: Can we call `r.b.GetLoad` directly instead of making a babysitter client?

File: internal/gke/nanny.go [hunk 2]
```diff
@@ -399,6 +409,57 @@ func getListenerPort(ctx context.Context, s store.Store, logger *slog.Logger, cl
...
+				wait.Done()
+			}()
+			babysitterAddr := fmt.Sprintf("https://%s:%d", ip, babysitterPort)
+			babysitter := newBabysitter(babysitterAddr)
+			reply, err := babysitter.GetLoad(ctx, &endpoints.GetLoadRequest{})
+			if err != nil {
+				logger.Debug("cannot get weavelet load: treating as unhealthy", "replica_set", replicaSet, "pod", pod.Name, "addr", babysitterAddr, "error", err)
+				return
+			}
+			mu.Lock()
```
Comment (mwhittaker) [inline internal/gke/nanny.go:451]: Is there a corresponding unlock?

**Review by mwhittaker — APPROVED** [review]
I still have to read through some of the tests, but the code changes look awesome! Just to make sure I'm understanding, this code simplifies things in two key ways: 

1. Before, replicas would register themselves, and the assigner would periodically health check them with a custom health checking protocol. Now, replicas don't register themselves, and health checking is done directly via Kubernetes.
2. We use `store.List` to avoid managing and traversing a bunch of keys.

**Review by mwhittaker — APPROVED** [review]

**Review by spetrovic77 — COMMENTED** [review]
Thanks for a thorough review!

**Pull request merged**